src/main/java/io/antmedia/security/TokenService.java

```diff
@@ -102,14 +107,42 @@ public void setSettings(AppSettings settings) {
 
 	public IDataStore getDataStore() {
 		if(dataStore == null) {
-			dataStore = ((DataStoreFactory) applicationContext.getBean("dataStoreFactory")).getDataStore();
+			dataStore = ((DataStoreFactory) applicationContext.getBean(IDataStoreFactory.BEAN_NAME)).getDataStore();
 		}
 		return dataStore;
 	}
 
 	public void setDataStore(IDataStore dataStore) {
 		this.dataStore = dataStore;
 	}
+	
+
+	public Map<String, String> getAuthenticatedMap() {
+		return authenticatedMap;
+	}
+
+	public void setAuthenticatedMap(Map<String, String> authenticatedMap) {
+		this.authenticatedMap = authenticatedMap;
+	}
+
+	@Override
+	public void sessionCreated(HttpSessionEvent se) {
+		logger.info("session created {}", se.getSession().getId());
+		
+	}
+
+
+	@Override
+	public void sessionDestroyed(HttpSessionEvent se) {
+		logger.info("session closed {}", se.getSession().getId());
+		
+		
+		if(!authenticatedMap.isEmpty() && authenticatedMap.containsKey(se.getSession().getId())) {
+			
+			authenticatedMap.remove(se.getSession().getId());
```

Comment:
what if we directly  call `authenticatedMap.remove(se.getSession().getId());`  and remove the if statement.

Reply:
ok